### PR TITLE
Hcl v.0.12

### PR DESCRIPTION
**Pull request by euliancom**
## Description
- refactoring class JitHelper to generate HCL instead of JSON
- refactoring default templates
- added templates for HCL2
- refactoring command convert

## Type of Changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a change to the documentation

## How Has This Been Tested?
- terrahub run -a -y

### Test Configuration
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have checked that my changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/helpers/jit-helper.js [hunk 1]
```diff
@@ -707,12 +707,58 @@ class JitHelper {
...
-      fse.mkdirsSync(tmpPath);
+  /**
+   * @param {Object} config
+   * @return {Boolean}
+   */
+  static checkTfVersion(config) {
+    const { terraform } = config;
+    if (terraform) {
+      const { version } = terraform;
+      const regExVersion = /0\.11/gm;
```
Comment (eistrati) [inline src/helpers/jit-helper.js]: this implementation works only tf v0.11, but doesn't work with v0.10 or v0.9 ... please use `semver` instead, see https://github.com/TerraHubCorp/terrahub/blob/master/bin/cli.js#L6

File: src/helpers/jit-helper.js [hunk 2]
```diff
@@ -707,12 +707,58 @@ class JitHelper {
...
+    let formatHCL1 = "";
+    if (!isHCL2) {
+      formatHCL1 = "-F no"
+    }
+
+    const arch = Downloader.getOsArch();
+    const componentBinPath = join(binPath, arch);
+
+    let extension = '';
+    if (arch.indexOf("windows") > -1)
```
Comment (vcalmic) [inline src/helpers/jit-helper.js]: Please wrap this `if` in curly braces

File: src/commands/convert.js [hunk 3]
```diff
@@ -84,22 +90,132 @@ class ConvertCommand extends TerraformCommand {
    * @private
    */
   _toJson(cfg) {
+    const currentFormat = this._checkCurrentFormat(cfg);
+    if (currentFormat != 'json') {
```
Comment (const-andreyLu) [inline src/commands/convert.js]: `!==` is safer

File: src/commands/convert.js [hunk 4]
```diff
@@ -84,22 +90,132 @@ class ConvertCommand extends TerraformCommand {
...
+   * @param {Object} cfg
+   * @return {Promise}
+   * @private
+   */
+  _toHCLFromJson(cfg) {    
     const componentConfigPath = join(ConvertCommand._buildComponentPath(cfg), config.defaultFileName);
     const rawConfig = ConfigLoader.readConfig(componentConfigPath);
 
+    if (rawConfig.component.hasOwnProperty('template') || this._checkIfFilesIsJson(cfg)) {
+      return ConvertCommand._saveComponent(cfg).then(() => {this.logSuccess(cfg.name, 'HCL'); });
```
Comment (const-andreyLu) [inline src/commands/convert.js]: space `{ this.logSuccess(cfg.name, 'HCL'); }`

File: src/helpers/jit-helper.js (hunk 2 above)
Comment (const-andreyLu) [inline src/helpers/jit-helper.js]: suggest
`const extension = arch.indexOf("windows") > -1 ? '.exe' : '';`

**Pull request merged**